rpc/block_svc_routines.c

```diff
@@ -1916,22 +1917,26 @@ glusterBlockReplaceNodeRemoteAsync(struct glfs *glfs, blockReplaceCli *blk,
   if (GB_ALLOC_N(args, info->mpath + 1) < 0) {
     goto out;
   }
-  args[0].glfs = glfs;
-  args[0].obj = (void *)cobj;
-  args[0].addr = blk->new_node;
-  args[0].volume = blk->volume;
   if (GB_STRDUP(cobj->block_hosts, blk->new_node) < 0) {
     goto out;
   }
   for (i = 0; i < info->nhosts; i++) {
     if (strcmp(info->list[i]->addr, blk->old_node)) {
       if (blockhostIsValid(info->list[i]->status)) {
+        /* Construct block_hosts */
         tmp = cobj->block_hosts;
         if (GB_ASPRINTF(&cobj->block_hosts, "%s,%s", tmp, info->list[i]->addr) == -1) {
           GB_FREE (tmp);
           goto out;
         }
         GB_FREE(tmp);
+
+        /* Get prio idx */
+        if (info->prio_path[0] && !strcmp(info->prio_path, info->list[i]->addr)) {
+          cobj->prio_idx = j;
```

Comment:
It is better to send the prio_path instead of prio index over the wire as it will be independent of the way info is constructed as of today. In future if there is a change in the way meta is constructed which changes the order of hosts then this code can lead to problem?

Reply:
Sure.